astropy/config/configuration.py

```diff
@@ -827,3 +777,63 @@ def update_default_config(pkg, default_cfg_dir_or_fn, version=None, rootname='as
             return True
 
     return False
+
+
+def create_config_file(pkg, rootname='astropy', overwrite=False):
+    """
+    Create the default configuration file for the specified package.
+    If the file already exists, it is updated only if it has not been
+    modified.  Otherwise the ``overwrite`` flag is needed to overwrite it.
+
+    Parameters
+    ----------
+    pkg : str
+        The package to be updated.
+    rootname : str
+        Name of the root configuration directory.
+    overwrite : bool
+        Force updating the file if it already exists.
+
+    Returns
+    -------
+    updated : bool
+        If the profile was updated, `True`, otherwise `False`.
+
+    """
+
+    # local import to prevent using the logger before it is configured
+    from astropy.logger import log
+
+    cfgfn = get_config_filename(pkg, rootname=rootname)
+
+    # generate the default config template
+    template_content = io.StringIO()
+    generate_config(pkg, template_content)
+    template_content.seek(0)
+    template_content = template_content.read()
+
+    doupdate = True
+
+    # if the file already exists, check that it has not been modified
+    if cfgfn is not None and path.exists(cfgfn):
+        with open(cfgfn, 'rt', encoding='latin-1') as fd:
+            content = fd.read()
+
+        identical = (content == template_content)
```

Comment:
Instead of comparing the config files byte-for-byte, why not just parse the existing config file and compare it to the default configuration?

This might require some refactoring of `generate_config` to have it generate the actual config data structure first, then a separate method for writing it to a file; probably how it should have been done in the first place.

Unless we really care about ensuring that it is byte-for-byte identical.

Reply:
Yes, that's the way it was done. But just after `is_unedited_config_file` loads the config file and checks if any keys are defined. So, since now this will not be run at import, maybe we can just keep the second check?